src/game_objects/grid.h

```diff
@@ -28,6 +28,11 @@ class Grid : public GameObject{
     std::mutex goal_mutex; // In case a new goal is created before old one is removed.
     int goal_z, goal_x;
 
+    // Preview constructs
+    Construct *preview; // the currently selected construct as a transparent preview on the grid
+    Construct *chest_preview;
```

Comment:
Again, working with the confines of existing code :(
Only the class Chest which derives from Construct knows how to set up the model, which I need as a preview.